specification/dartLangSpec.tex

```diff
@@ -10474,84 +10508,158 @@ \subsection{Assignment}
 \LMLabel{assignment}
 
 \LMHash{}%
-An assignment changes the value associated with a mutable variable or property.
+An assignment changes the value associated with a mutable variable,
+or invokes a setter.
 
 \begin{grammar}
 <assignmentOperator> ::= `='
   \alt <compoundAssignmentOperator>
+
+<compoundAssignmentOperator> ::= `*='
+  \alt `/='
+  \alt `~/='
+  \alt `\%='
+  \alt `+='
+  \alt `-='
+  \alt `\ltlt='
+  \alt `\gtgt='
+  \alt `\gtgtgt='
+  \alt `\&='
+  \alt `^='
+  \alt `|='
+  \alt `??='
 \end{grammar}
 
 \LMHash{}%
-\Case{\code{$v$ = $e$}}
-%% TODO(eernst): This _only_ works if we assume that `v = e` has already
-%% been expanded to `this.v = e` "when that's the right thing to do".
-%% Otherwise `denotes` below cannot be interpreted as the result of a lookup,
-%% and we have no precise alternative which would work. We might be able
-%% to repair this by giving a definition of `denotes` somewhere.
-Consider an assignment $a$ of the form \code{$v$ = $e$},
-where $v$ is an identifier or an identifier qualified by an import prefix,
-and $v$ denotes a variable (\ref{variables}) or \code{$v$=} denotes a setter
-(\commentary{which may be declared explicitly or induced by an instance variable, etc}).
-Let $T$ be the static type of $v$ when $v$ denotes a variable,
-otherwise let $T$ be the static type of the formal parameter of the setter \code{$v$=}.
-It is a compile-time error if the static type of $e$ may not be assigned to $T$.
-The static type of $a$ is the static type of $e$.
+\Case{\code{\id{} = $e$}}
+Consider an assignment $a$ of the form \code{\id{} = $e$},
+where \id{} is an identifier.
+
+If \id{} is a local variable $v$
```

Comment:
Why not talk about the result of lexical lookup here? How do you *know* that *id* "is a local variable" without doing lexical lookup.

Reply:
Done.